include/cctz/civil_time_detail.h

```diff
@@ -21,7 +21,7 @@
 #include <type_traits>
 
 // Disable constexpr support unless we are in C++14 mode.
-#if __cpp_constexpr >= 201304 || _MSC_VER >= 1910
+#if __cpp_constexpr >= 201304 || (defined(_MSC_VER) && _MSC_VER >= 1910)
```

Comment:
I'm in two minds about this.

On the one hand, the pre-processor behavior is well defined and lots of code depends on it.  That is, I'm not a fan of `-Wundef`.

On the other hand, if we're going to be explicit, we should be so consistently.  For example, there is another instance of the same issue on that very line.

Thoughts?